tests/providers/yandex/hooks/test_yandex.py

```diff
@@ -58,7 +59,10 @@ def test_provider_user_agent(self, mock_get_credentials, mock_get_connection):
         mock_get_credentials.return_value = {"token": 122323}
         sdk_prefix = "MyAirflow"
 
-        with conf_vars({("yandex", "sdk_user_agent_prefix"): sdk_prefix}):
+        with conf_vars({("yandex", "sdk_user_agent_prefix"): sdk_prefix}), pytest.warns(
+            AirflowProviderDeprecationWarning,
+            match="Using `provider_user_agent` in `YandexCloudBaseHook` is deprecated. Please use it in `utils.user_agent` instead.",
+        ):
             hook = YandexCloudBaseHook()
```

Comment:
It's better to move the hook init. out of the context manager:
```suggestion
        hook = YandexCloudBaseHook()
        with conf_vars({("yandex", "sdk_user_agent_prefix"): sdk_prefix}), pytest.warns(
            AirflowProviderDeprecationWarning,
            match="Using `provider_user_agent` in `YandexCloudBaseHook` is deprecated. Please use it in `utils.user_agent` instead.",
        ):
```